### PR TITLE
Added push gateway

### PR DESCRIPTION
**Pull request by goruha**
## What
* Create prometheus push gateway

## Why
* To collect metrics from short live processes

### CODE REVIEW

File: helmfile.d/0400.kube-prometheus.yaml [hunk 1]
```diff
@@ -159,6 +159,74 @@ releases:
...
+## kube-prometheus-push-gateway                                                      ##
+## Provide kube-prometheus default push gateway                                      ##
+#######################################################################################
+
+#
+# References:
+#   - https://github.com/coreos/prometheus-operator/tree/master/helm/kube-prometheus
+#   - https://github.com/coreos/prometheus-operator/tree/master/contrib/kube-prometheus
+#
+- name: "kube-prometheus-pushgateway-scraper"
```
Comment (osterman) [inline helmfile.d/0400.kube-prometheus.yaml]: why scraper? it doesn't scrape.

File: helmfile.d/0400.kube-prometheus.yaml [hunk 2]
```diff
@@ -159,6 +159,74 @@ releases:
...
+  namespace: "monitoring"
+  labels:
+    chart: "prometheus-pushgateway"
+    component: "monitoring"
+    namespace: "monitoring"
+    vendor: "cloudposse"
+    default: "true"
+  chart: "cloudposse-incubator/monochart"
```
Comment (osterman) [inline helmfile.d/0400.kube-prometheus.yaml:207]: monochart FTW!

File: helmfile.d/0400.kube-prometheus.yaml (hunk 1 above)
Comment (osterman) [reply helmfile.d/0400.kube-prometheus.yaml]: Oh, I see. This is defining the service monitors. I think maybe we should use (standardize around) `-monitors`

Comment (goruha) [reply helmfile.d/0400.kube-prometheus.yaml]: It provides ServiceMonitor that tells prometheus what to scape

Comment (goruha) [reply helmfile.d/0400.kube-prometheus.yaml]: ok

File: helmfile.d/0400.kube-prometheus.yaml [hunk 3]
```diff
@@ -159,6 +159,71 @@ releases:
 {{ end }}
 
 
+#######################################################################################
+## kube-prometheus-pushgateway                                                       ##
+## Provide kube-prometheus default push gateway                                      ##
```
Comment (aknysh) [inline helmfile.d/0400.kube-prometheus.yaml]: Provides

File: helmfile.d/0400.kube-prometheus.yaml [hunk 4]
```diff
@@ -159,6 +159,71 @@ releases:
...
+  version: "0.1.2"
+  wait: true
+  values:
+  - serviceAccount:
+      create: {{ env "KUBE_PROMETHEUS_SERVICE_ACCOUNT_CREATE" | default "false" }}
+    replicaCount: {{ env "KUBE_PROMETHEUS_PUSH_GATEWAY_REPLICA_COUNT" | default "1" }}
+
+#######################################################################################
+## kube-prometheus-pushgateway-monitor                                               ##
+## Provide kube-prometheus service monitor for  push gateway                         ##
```
Comment (aknysh) [inline helmfile.d/0400.kube-prometheus.yaml]: Provides

File: helmfile.d/0400.kube-prometheus.yaml [hunk 5]
```diff
@@ -159,6 +159,71 @@ releases:
...
+
+#
+# References:
+#   - https://github.com/coreos/prometheus-operator/tree/master/helm/kube-prometheus
+#   - https://github.com/coreos/prometheus-operator/tree/master/contrib/kube-prometheus
+#
+- name: "kube-prometheus-pushgateway-monitor"
+  namespace: "monitoring"
+  labels:
+    chart: "prometheus-pushgateway"
```
Comment (aknysh) [inline helmfile.d/0400.kube-prometheus.yaml:202]: why it's the same `chart: "prometheus-pushgateway"` as above?

File: helmfile.d/0400.kube-prometheus.yaml (hunk 3 above)
Comment (goruha) [reply helmfile.d/0400.kube-prometheus.yaml]: Done

File: helmfile.d/0400.kube-prometheus.yaml (hunk 4 above)
Comment (goruha) [reply helmfile.d/0400.kube-prometheus.yaml]: Done

**Review by aknysh — COMMENTED** [review]
see comments

File: helmfile.d/0400.kube-prometheus.yaml (hunk 5 above)
Comment (goruha) [reply helmfile.d/0400.kube-prometheus.yaml:202]: `monochart` is too common. 
By mean this release is addition to `prometheus-pushgateway` that intergrates it with `kube-prometheus` and should be installed togeather

**Comment by goruha** [comment]
@aknysh  Done

**Review by osterman — APPROVED** [review]

**Pull request merged**